tests/test_dataset_mask.py

```diff
@@ -176,6 +191,17 @@ def test_kwargs(tiffs):
         other = src.dataset_mask(window=((1, 4), (1, 4)), boundless=True)
         assert np.array_equal(alp_shift_lr, other)
 
+        other = src.dataset_mask(out_shape=(1, 5, 5))
+        assert np.array_equal(resampmask, other)
+
+        out = np.zeros((1, 5, 5), dtype=np.uint8)
+        other = src.dataset_mask(out=out)
+        assert np.array_equal(resampmask, other)
+
+        other = src.dataset_mask(out_shape=(1, 5, 5), resampling=Resampling.average) != 0
+        other = other.astype(np.uint8) * 255
+        assert np.array_equal(resampave, other)
```

Comment:
@vincentsarago can you split lines 201-203 out into their own test?

This new test fails with GDAL < 2.2 and I think it's possible that the averaging algorithm changed between GDAL versions. Let's test with the bilinear algorithm, which is more stable.